salt/modules/salt_check.py

```diff
@@ -0,0 +1,449 @@
+            search_list.append(path)
+        path = cachedir + os.sep + "files" + os.sep + "base"
+        search_list.append(path)
+        return search_list
+
+    def get_state_dir(self):
+        ''''return the path of the state dir'''
+        paths = self.get_state_search_path_list()
+        return paths
+
+
+class StateTestLoader(object):
+    '''
+    Class loads in test files for a state
+    e.g.  state_dir/salt-check-tests/[1.tst, 2.tst, 3.tst]
+    '''
+
+    def __init__(self, search_paths):
+        self.search_paths = search_paths
+        self.path_type = None
+        self.test_files = []  # list of file paths
+        self.test_dict = {}
+
+    def load_test_suite(self):
+        '''load tests either from one file, or a set of files'''
+        for myfile in self.test_files:
+            self.load_file(myfile)
+
+    def load_file(self, filepath):
+        '''
+        loads in one test file
+        '''
+        try:
+            myfile = open(filepath, 'r')
+            contents_yaml = yaml.load(myfile)
+            for key, value in contents_yaml.items():
+                self.test_dict[key] = value
+        except:
+            raise
+        return
+
+    def gather_files(self, filepath):
+        '''gather files for a test suite'''
+        filepath = filepath + os.sep + 'salt-check-tests'
+        rootDir = filepath
+        for dirName, subdirList, fileList in os.walk(rootDir):
+            for fname in fileList:
+                if fname.endswith('.tst'):
+                    start_path = dirName + os.sep + fname
+                    full_path = os.path.abspath(start_path)
+                    self.test_files.append(full_path)
+        return
+
+    def find_state_dir(self, state_name):
+        '''find and return the path to the state dir'''
+        state_path = None
+        for path in self.search_paths:
+            rootDir = path
+            for dirName, subdirList, fileList in os.walk(rootDir):
+                mydir = dirName.split(os.sep)[-1]
+                if state_name == mydir:
+                    state_path = dirName
+                    return state_path
+        return state_path
+
+
+def _get_test_files(state_name):
+    '''Given a path to the state files, gather the list of test files under
+    the salt-check-test subdir'''
+    salt_check = SaltCheck()
+    paths = salt_check.get_state_search_path_list()
+    stl = StateTestLoader(search_paths=paths)
+    mydir = stl.find_state_dir(state_name)
+    stl.gather_files(mydir)
+    return stl.test_files
+
+
+def _get_top_states():
+    ''' Show the dirs for the top file used for a particular minion'''
+    salt_check = SaltCheck()
+    return salt_check.get_top_states()
+
+
+def run_state_tests(state_name):
+    '''
+    Runs tests for one state
+    CLI Example:
+        salt '*' salt_check.run_state_tests STATE-NAME
+    '''
+    if not state_name:
+        return "State name required"
+    scheck = SaltCheck()
```

Comment:
if it is not up above it certinly should be here, should be able to send the opts in as a __init__ argument